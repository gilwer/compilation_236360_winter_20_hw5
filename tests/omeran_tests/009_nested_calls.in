// nested calls, change variables and arguments
enum days { SUN, MON, TUES };

void foo(int a) {
  a = a + 1;
  print("in foo a is");
  printi(a);
}

void bar(int kk) {
  print("in bar kk is");
  printi(kk);
  foo(kk);
  print("after calling foo in bar with kk");
  printi(kk);
  int bb = 3;
  print("in bar bb is");
  printi(bb);
  foo(bb);
  print("after calling foo in bar with bb");
  printi(bb);
}

void main() {
  bar(7); // 7, 8, 7, 3, 4 ,3
}