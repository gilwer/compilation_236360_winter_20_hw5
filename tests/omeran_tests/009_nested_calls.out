in bar kk is
7
in foo a is
8
after calling foo in bar with kk
7
in bar bb is
3
in foo a is
4
after calling foo in bar with bb
3
